cirq-core/cirq/ops/gateset.py

```diff
@@ -225,17 +233,33 @@ def __init__(
             name: (Optional) Name for the Gateset. Useful for description.
             unroll_circuit_op: If True, `cirq.CircuitOperation` is recursively
                 validated by validating the underlying `cirq.Circuit`.
-            accept_global_phase_op: If True, `cirq.GlobalPhaseOperation` is accepted.
+            accept_global_phase_op: If True, a `GateFamily` accepting
+                `cirq.GlobalPhaseGate` will be included. If None,
+                `cirq.GlobalPhaseGate` will still be accepted if it appears in
+                the `gates` parameter. If False, `cirq.GlobalPhaseGate` will be
+                removed from the gates. This parameter defaults to False (a
+                breaking change from v0.14) and will be removed in v0.16.
         """
         self._name = name
         self._unroll_circuit_op = unroll_circuit_op
-        self._accept_global_phase_op = accept_global_phase_op
+        if accept_global_phase_op:
+            gates = gates + (global_phase_op.GlobalPhaseGate,)
         self._instance_gate_families: Dict[raw_types.Gate, GateFamily] = {}
         self._type_gate_families: Dict[Type[raw_types.Gate], GateFamily] = {}
         self._gates_repr_str = ", ".join([_gate_str(g, repr) for g in gates])
         unique_gate_list: List[GateFamily] = list(
             dict.fromkeys(g if isinstance(g, GateFamily) else GateFamily(gate=g) for g in gates)
         )
+        if accept_global_phase_op is False:
+            unique_gate_list = [
+                g for g in unique_gate_list if g.gate is not global_phase_op.GlobalPhaseGate
+            ]
+        elif accept_global_phase_op is None:
+            if not any(g.gate is global_phase_op.GlobalPhaseGate for g in unique_gate_list):
+                warnings.warn(
+                    'v0.14.1 is the last release `cirq.GlobalPhaseGate` is included by default.'
```

Comment:
Please make this message more descriptive. The warning will be triggered in two cases:

1. The user wants the gateset to accept global phase gates but relied on the default value of the parameter. The warning should clearly tell this user to explicitly include a `cirq.GlobalPhaseGate` to the list of gates passed to the constructor (and as a result, the warning will go away). 
2. The user does not want to accept global phase gates, and hence didn't explicitly pass a `cirq.GlobalPhaseGate` to the constructor. To avoid a deprecation message, the user also didn't explicitly set `accept_global_phase_op = False` in the constructor. The warning should convey to this user that they should just ignore this warning and it will go away automatically in the next release. 

I wish we could handle case-2 better, but I don't think we have much of an option here.

Reply:
done